test/python/circuit/test_circuit_operations.py

```diff
@@ -55,32 +55,39 @@ def test_combine_circuit_common(self):
         qc1.h(qr[0])
         qc1.measure(qr[0], cr[0])
         qc2.measure(qr[1], cr[1])
-        new_circuit = qc1 + qc2
+
+        new_circuit = qc1.combine(qc2)
+
         backend = BasicAer.get_backend('qasm_simulator')
         shots = 1024
         result = execute(new_circuit, backend=backend, shots=shots, seed_simulator=78).result()
         counts = result.get_counts()
         target = {'00': shots / 2, '01': shots / 2}
         threshold = 0.04 * shots
+        self.assertDictEqual(qc1.count_ops(), {'h': 1, 'measure': 1})  # no changes "in-place"
+        self.assertDictEqual(qc2.count_ops(), {'measure': 1})  # no changes "in-place"
         self.assertDictAlmostEqual(counts, target, threshold)
 
-    def test_combine_circuit_different(self):
-        """Test combining two circuits with different registers.
+    def test_combine_circuit_common_plus(self):
+        """Test combining two circuits with same registers (as plus).
         """
         qr = QuantumRegister(2)
         cr = ClassicalRegister(2)
-        qc1 = QuantumCircuit(qr)
-        qc1.x(qr)
+        qc1 = QuantumCircuit(qr, cr)
```

Comment:
Are the changes to `qc1` here still needed (since the behavior of `+` is not changing in this PR)?